### ISSUE TITLE
update allele name automatically if unknown/known combination

### ISSUE THREAD

**Issue by ValWood** [pombase/pombase-chado#874]
I think 
 SPBC146.07	prp2-2	190db567d6fe649a	T259I	prp2-T259I	d5d0ef3568cea1fd

https://www.pombase.org/genotype/prp2-C387Y-C387Y-amino_acid_mutation-expression-not_assayed
https://www.pombase.org/genotype/prp2-1-C387Y-amino_acid_mutation-expression-not_assayed

creates separate allele pages.

I thought we had a rule that
if an allele was type  unknown, as in 
 SPBC146.07	prp2-2	190db567d6fe649a

but then changed to a specific type (i.e amino acid mutation)
T259I	prp2-T259I	d5d0ef3568cea1fd

the allele description would automagically update/merrge in some way.

If  this never happened, then it probably should.
IN these cases I think the prp2-T259I is more informative as a name, and prp2-2 could be a synonym.

**Comment by ValWood** [comment]
It struck me that the automagical conversion might not be possible if we already  have:
prp2-T259I(T259I)
AND
prp2-2(T259I)

**Comment by mah11** [comment]
The follow-up comment is accurate -- the automagic conversion only kicks in when the allele type is `unknown` (and is working, as far as I can tell). In the prp2 example, the C387Y allele has two different names, and get flagged by the `alleles with the same description that have different names` check in the `chado_checks` log.

**Comment by mah11** [comment]
also, prp2-2 is not `unknown` in 190db567d6fe649a

**Comment by ValWood** [comment]
>also, prp2-2 is not unknown in 190db567d6fe649a

No, because I fixed it. But it *was* earlier today.
See the description field in the log file above is empty.

**Comment by mah11** [comment]
Yes, `unknown` conversion is working - for example, prp2-2 is `unknown` in the session for PMID:11350031 (24a3853610543a99), and that paper is included on the prp2-C387Y(aa) genotype page.

**Comment by ValWood** [comment]
Right I was looking at prp2-1.

I'm stilll a bit confused. So the conversion happens but the difference is still reported in the logs , is that it?

Anyway this one can close.

**Comment by mah11** [comment]
No, if the allele type is `unknown` it gets converted to a "known" one. Two different names for the same "known" description are reported in the `chado_checks` log. Neither is converted to the other.

The slightly confusing thing about prp2-1 is that it has both things going on - there are sessions where its type is `unknown` AND at least one session where it has another name, prp2-C378Y.

(prp2-2 was a typo in my other comment; now fixed)
